adding debugging details to migration overview

SharpRake opened the pull request:
## Type of change
<!-- Please be sure to add the appropriate label to your PR. -->
This adds some debugging details to the recently-published migration overview doc.

### What should this PR do?
<!-- Does this PR resolve an issue? Please include a reference to it. -->
Resolves https://github.com/chainguard-dev/internal/issues/4043

### Why are we making this change?
<!-- What larger problem does this PR address? -->

### What are the acceptance criteria? 
<!-- What should be happening for this PR to be accepted? Please list criteria. -->
<!-- Do any stakeholders need to be tagged in this review? If so, please add them. -->

### How should this PR be tested?
<!-- What should your reviewer do to test this PR? Please list steps. -->
Please review and ensure that it makes sense. Also, if there are other details I could add (without this update getting overly long) please let me know!

SharpRake committed:
adding debugging details to migration overview

amouat commented on content/chainguard/migration/migrations-overview.md:
```diff
@@ -290,9 +290,57 @@ In addition, we have a few migration guides in the form of videos:
 * [Ruby](/chainguard/chainguard-images/getting-started/ruby/)
 
-## Troubleshooting Resources
+## Common Pitfalls
```
I think this might need to be merged with the tips section. There's a lot of overlap.

SharpRake replied:
fair point, I was rushing to get this off my plate and wasn't thinking of how well it all fit

amouat commented on content/chainguard/migration/migrations-overview.md:
````diff
@@ -290,9 +290,57 @@ In addition, we have a few migration guides in the form of videos:
+In cases like this, you can instead run a command like the following to access the container's shell as the root user:
+
+```sh
+docker run -it -u 0:0 --entrypoint bin/bash chainguard/python:latest-dev
````
`--user root` will work as well

SharpRake replied:
That's more straightforward, thanks!

amouat commented on content/chainguard/migration/migrations-overview.md:
```diff
@@ -290,9 +290,57 @@ In addition, we have a few migration guides in the form of videos:
+
+### Copying in 3rd Party Binaries
+
+You may have Docker builds that copy in binaries to run agents or similar tooling. In some cases you may find these binaries don’t work as expected as they are designed to run on a different Linux distribution.
```
Maybe add "Chainguard Images may not have the dependencies required by 3prd party binaries, or they may be stored at a different path."

amouat reviewed (changes requested):
The content is fine, but I think it needs to either be on a different page or merged with the tips section. As it stands there is a lot of overlap which is a confusing for the reader.

SharpRake committed:
AM's suggestions, + removing a line from another doc

SharpRake commented:
@amouat Thanks for the feedback! I've moved things around a bit: I pulled the new info up into the tips section, deleted the redundant paragraph about images not having shells, and combined the "copying in 3rd party binaries" section into the "packages not found" section.

Let me know if you think anything could be improved further!

SharpRake commented:
Oh, also I removed a line about `wolfi-base` in another doc per [this slack discussion](https://chainguard-dev.slack.com/archives/C04PYHWPE1F/p1722274552041219).

mcaveety commented:
> Oh, also I removed a line about `wolfi-base` in another doc per [this slack discussion](https://chainguard-dev.slack.com/archives/C04PYHWPE1F/p1722274552041219).

This looks okay, I will note that I included that extra line because I only listed a portion of the complete packages. I am quite lost from the Slack thread so I'll defer 🫠

amouat commented:
Did you forget to push something? I think it's still separate sections and there's still the para on dev images.
